src/main/java/de/hpi/bpt/argos/persistence/model/parsing/BackboneDataParserImpl.java

```diff
@@ -0,0 +1,448 @@
+				prediction *= TO_PERCENT;
+			}
+		} catch (Exception e) {
+			logger.debug(String.format("can not parse cause prediction '%1$s'", causePrediction));
+			logTrace(e);
+			return;
+		}
+
+		ErrorCause cause = new ErrorCauseImpl();
+		cause.setDescription(tempCurrentCauseDescription);
+		cause.setErrorPrediction(prediction);
+		currentErrorType.addErrorCause(cause);
+
+		errorCausesImported++;
+	}
+
+	/**
+	 * This method saves the finished product in the database.
+	 */
+	protected void saveCurrentProduct() {
+
+		if (currentProduct == null) {
+			return;
+		}
+
+		logger.info(String.format("added product '%1$s' with '%2$d' configurations in family '%3$s'", currentProduct.getName(), currentProduct
+				.getProductConfigurations().size(), currentProduct.getProductFamily().getName()));
+		entityManager.updateEntity(currentProduct, ProductEndpoint.getProductUri(currentProduct.getId()));
+
+		resetCurrentEntities();
+	}
+
+	/**
+	 * This method resets all temporary variables.
+	 */
+	protected void resetCurrentEntities() {
+
+		tempCurrentProductExternalId = "";
+		tempCurrentProductFamilyId = "";
+		tempCurrentCauseDescription = "";
+		tempCurrentProductName = "";
+		tempCurrentCauseCode = "";
+		tempCurrentErrorDisplayCode = "";
+		tempCurrentCodingPlug = "";
+		currentProduct = null;
+		currentErrorType = null;
+		currentProductConfiguration = null;
+	}
+
+	/**
+	 * This method checks, whether a error type is already present in the current product.
+	 * @param errorType - the error type to check
+	 * @return - true, if the error type is already present
+	 */
+	protected boolean currentProductConfigurationHasErrorType(ErrorType errorType) {
+		if (currentProductConfiguration == null) {
+			return false;
+		}
+
+		for (ErrorType type : currentProductConfiguration.getErrorTypes()) {
+			if (type.getErrorTypeId().equalsIgnoreCase(errorType.getErrorTypeId())) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+    /**
+     * Logs an exception on log level trace.
+     * @param e - exception to log
+     */
+	private void logTrace(Exception e) {
+		logger.trace("Reason: ", e);
+    }
+
+	/**
+	 * This method logs the statistics about the imported entities.
+	 */
+	protected void logStatistics() {
+		logger.info(String.format("finished importing data from '%1$s' in '%2$s'. "
+				+ "imported %3$d products, "
+				+ "%4$d productConfigurations, "
+				+ "%5$d errorTypes and "
+				+ "%6$d errorCauses.",
+				fileName,
+				getTimeSinceStart(),
+				productsImported,
+				productConfigurationsImported,
+				errorTypesImported,
+				errorCausesImported));
+	}
+
+	/**
+	 * This method calculates the passed time since the import process started and returns it as a string.
+	 * @return - the passed time since start of the import process as string
+	 */
+	protected String getTimeSinceStart() {
+
+		long passedTimeInSeconds = ((new Date()).getTime() - importStartTime.getTime()) / 1000;
+		long[] passedTime = new long[] { 0, 0, 0, 0 };
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') '{' is followed by whitespace. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=checkstyle%3ANo_Whitespace_After_BP2016)